src/components/LibraryMenuHeaderContent.tsx

```diff
@@ -293,14 +295,24 @@ export const LibraryDropdownMenu = ({
       library.setLibrary(nextItems).catch(() => {
         setAppState({ errorMessage: t("alerts.errorRemovingFromLibrary") });
       });
+
+      deleteItemsFromLibraryCache(selectedItems);
+
       onSelectItems([]);
     },
-    [library, setAppState, selectedItems, onSelectItems],
+    [
+      library,
+      setAppState,
+      selectedItems,
+      onSelectItems,
+      deleteItemsFromLibraryCache,
+    ],
   );
 
   const resetLibrary = useCallback(() => {
     library.resetLibrary();
-  }, [library]);
+    clearLibraryCache();
+  }, [library, clearLibraryCache]);
```

Comment:
`clearLibraryCache` is not memoized

Reply:
but the component this is passed into isn't memoized itself, so memoizing the callback has no effect.

I'd remove the callback memoization, since the component it's passed into is not heavy.

That applies also to the other relevant callbacks (which btw also depend on non-memoized functions).